tests/Php84/ReflectionConstantTest.php

```diff
@@ -0,0 +1,229 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Polyfill\Tests\Php84;
+
+use Exception;
+use PHPUnit\Framework\TestCase;
+use ReflectionConstant;
+use ReflectionException;
+
+// Used by the test
+const EXAMPLE = 'Foo';
+
+/**
+ * @author Daniel Scherzer <daniel.e.scherzer@gmail.com>
+ */
+class ReflectionConstantTest extends TestCase
+{
+
+    public function testMissingConstant()
+    {
+        $this->expectException( ReflectionException::class );
+        $this->expectExceptionMessage(
+            'Constant "missing" does not exist'
+        );
+        $r = new ReflectionConstant( 'missing' );
+    }
+
+    public function testClassConstant()
+    {
+        $this->expectException( ReflectionException::class );
+        $this->expectExceptionMessage(
+            'Constant "ReflectionClass::IS_DEPRECATED" does not exist'
+        );
+        // Confirm that this constant actually exists
+        $this->assertTrue( defined( 'ReflectionFunction::IS_DEPRECATED' ) );
```

Comment:
this assertion should come *before* the configuration of expecting an exception (as we don't want want to accept such exception during that statement)